compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentCppWriter.scala

```diff
@@ -288,42 +288,48 @@ case class ComponentCppWriter (
   }
 
   private def getBuffUnion: List[Line] = {
-    line("// Get the max size by doing a union of the input and internal port serialization sizes") ::
+    // Collect the serialized sizes of all the async port arguments
+    val portMembers = List.concat(
+      // Typed async input ports
+      typedAsyncInputPorts.flatMap(p => {
+        val portName = p.getUnqualifiedName
+        val portTypeName = getQualifiedPortTypeName(p, p.getDirection.get)
+        lines(s"BYTE ${portName}PortSize[${portTypeName}::SERIALIZED_SIZE];")
+      }),
+      // Command input port
+      guardedList (cmdRecvPort.isDefined) (lines(s"BYTE cmdPortSize[Fw::InputCmdPort::SERIALIZED_SIZE];")),
+      // Internal ports
+      internalPorts.flatMap(p =>
+        line(s"// Size of ${p.getUnqualifiedName} argument list") ::
```

Comment:
I followed the code up to this point.  Not sure what this is doing, nor why it doesn't look like the async block above.